app/static/src/app/store/sessions/actions.ts

```diff
@@ -26,13 +26,19 @@ export const actions: ActionTree<SessionsState, AppState> = {
             .get(url);
     },
 
-    // We don't use sessionId yet!
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     async [SessionsAction.Rehydrate](context, sessionId: string) {
-        // TODO: fetch data from session data endpoint, populate the current state and re-run models as necessary
-        // For this POC, just set some hardcoded value as the new session code
-        const placeholderCode = ["# Code for rehydration!", "initial(S) <- N - I_0"];
-        const { commit } = context;
-        commit(`code/${CodeMutation.SetCurrentCode}`, placeholderCode, { root: true });
+        // TODO: complete rehydrate of entire state, just setting code for now
+        const { rootState, dispatch } = context;
+        const { appName } = rootState;
+        const url = `/apps/${appName}/sessions/${sessionId}`;
+        const response = await api(context)
+            .ignoreSuccess()
+            .withError(`errors/${ErrorsMutation.AddError}`, true)
```

Comment:
Will this error handler be called? This would be in the case of a failure requesting the session which would be hard to trigger in practice?

Reply:
Yeah, should only really happen if there's network problems or the backend goes down. This is a generic error handler which wil show an error on screen.